OpenMarket/OpenMarket/Network/Network.swift

```diff
@@ -0,0 +1,119 @@
+        "https://market-training.yagom-academy.kr/"
+    }
+    
+    private static var session: URLSession {
+        let config = URLSessionConfiguration.default
+        config.waitsForConnectivity = true
+        config.timeoutIntervalForResource = 300
+        return URLSession(configuration: config)
+    }
+    
+    private var urlString: String {
+        switch self {
+        case .serverState:
+            return Self.host + "healthChecker"
+        case .requestList(let page, let itemsPerPage):
+            return Self.host + "api/products?items_per_page=\(itemsPerPage)&page_no=\(page)"
+        case .requestProduct(let id):
+            return Self.host + "api/products/\(id)"
+        }
+    }
+}
+
+//MARK: - Network Method
+
+extension API {
+    func checkServerState(session: URLSessionProtocol = session, completion: @escaping (Result<String, NetworkErorr>) -> Void) {
+        guard let url = URL(string: urlString) else {
+            completion(.failure(.urlError))
+            return
+        }
+        
+        session.dataTask(with: url) { data, response, error in
+            guard error == nil else {
+                completion(.failure(.severError))
+                return
+            }
+            
+            guard let statusCode = (response as? HTTPURLResponse)?.statusCode, (200..<300).contains(statusCode) else {
+                completion(.failure(.severError))
+                return
+            }
+            
+            guard let data = data, let text = String(data: data, encoding: .utf8) else {
+                completion(.failure(.unknown))
+                return
+            }
+            
+            completion(.success(text.trimmingCharacters(in:CharacterSet(charactersIn: "\""))))
+        }.resume()
+    }
+    
+    func request(session: URLSessionProtocol = session, completion: @escaping (Result<T, NetworkErorr>) -> Void) {
+        guard let url = URL(string: urlString) else {
+            completion(.failure(.urlError))
+            return
+        }
+        
+        session.dataTask(with: url) { data, response, error in
+            guard error == nil else {
+                completion(.failure(.severError))
+                return
+            }
+            
+            guard let responseCode = (response as? HTTPURLResponse)?.statusCode, (200..<300).contains(responseCode) else {
+                completion(.failure(.severError))
+                return
+            }
+            
+            guard let data = data else {
+                completion(.failure(.unknown))
+                return
+            }
+            
+            do {
+                let jsonDecoder = JSONDecoder()
+                jsonDecoder.keyDecodingStrategy = .convertFromSnakeCase
+                jsonDecoder.dateDecodingStrategy = .formatted(.dateFormatter)
+                
+                let productDetail = try jsonDecoder.decode(T.self, from: data)
+                completion(.success(productDetail))
```

Comment:
request는 어떤 API가 와도 전역적으로 사용될 수 있으면 좋겠어요:) 그래서 네이밍에서도 포괄적인 용어를 사용하는건 어떨까요?
예를 들어,
```suggestion
                let result = try jsonDecoder.decode(T.self, from: data)
                completion(.success(result))
```

Reply:
네이밍을 수정하다는 것을 까먹었네요. 올바르게 수정했습니다.